pyradiosky/skymodel.py

```diff
@@ -1696,21 +1722,167 @@ def from_recarray(
 
         return self
 
+    def read_skyh5(
+        self,
+        filename,
+        run_check=True,
+        check_extra=True,
+        run_check_acceptability=True,
+    ):
+        """
+        Read a skyh5 file (our flavor of hdf5) into this object.
+
+        Parameters
+        ----------
+        filename : str
+            Path and name of the skyh5 file to read.
+        run_check : bool
+            Option to check for the existence and proper shapes of parameters
+            after downselecting data on this object (the default is True,
+            meaning the check will be run).
+        check_extra : bool
+            Option to check optional parameters as well as required ones (the
+            default is True, meaning the optional parameters will be checked).
+        run_check_acceptability : bool
+            Option to check acceptable range of the values of parameters after
+            downselecting data on this object (the default is True, meaning the
+            acceptable range check will be done).
+
+        """
+        with h5py.File(filename, "r") as fileobj:
+            if "/Header" not in fileobj:
+                raise ValueError(
+                    "This is an old 'healvis' style healpix HDF5 file. To read it, "
+                    "use the `read_healpix_hdf5` method. Support for this file format "
+                    "is deprecated and will be removed in version 0.3.0."
+                )
+
+        with h5py.File(filename, "r") as fileobj:
+
+            # extract header information
+            header = fileobj["/Header"]
+            header_params = [
+                "_Ncomponents",
+                "_Nfreqs",
+                "_component_type",
+                "_spectral_type",
+                "_ra",
+                "_dec",
+                "_history",
+                "_name",
+                "_nside",
+                "_hpx_inds",
+                "_freq_array",
+                "_reference_frequency",
+                "_spectral_index",
+                "_beam_amp",
+                "_extended_model_group",
+            ]
+
+            optional_params = [
+                "_name",
+                "_nside",
+                "_hpx_inds",
+                "_freq_array",
+                "_reference_frequency",
+                "_spectral_index",
+                "_beam_amp",
+                "_extended_model_group",
+            ]
+            init_params = {}
+
+            for par in header_params:
+                param = getattr(self, par)
+                parname = param.name
+
+                # skip optional params if not present
+                if par in optional_params:
+                    if parname not in header:
+                        continue
+
+                dset = header[parname]
+                value = dset[()]
+
+                if "unit" in dset.attrs:
+                    value *= units.Unit(dset.attrs["unit"])
```

Comment:
I've seen this type of unit assignment fail in weird ways before `value *= unit` but if it is working then no big deal here :+1:

Reply:
hmm. Do you have a more robust way to do it?

Reply:
My go to is usually `value = value * unit` I know it looks silly because they are the same thing but there are some edge cases where the unit assignment can't be coerced or something. I think we don't need to change it, but we at least know a solution if the problem arises.

Reply:
ok, good to know.